_posts/2020-04-07-trying-out-julia.md

```diff
@@ -0,0 +1,227 @@
+---
+layout: post
+title: Trying out Julia programming language
+author: hnybom
+excerpt: >
+  Julia is a programming language with it's roots in the mathematics. The language itself is increasingly gaining momentum and that combined with the boom were seeing in data science demand makes it a very interesting subject. This post is about my journey starting from zero knowledge to, well, some knowledge of the language. So if you, like me, are interested in trying out Julia language keep on reading. If you are already familiar with the subject you can safely skip this one.
+tags:
+ - Julia
+---
+
+## Let's try out Julia
+
+As forewords, I have never tried Julia programming language before but I've read couple of headlines about it and they have peaked my interest on the subject. The main motivations to get into the language for me are
+
+* Julia seems to be gaining some momentum in both the programming and especially data science community.
+* Even as Julia is dynamically typed it has some static typing going on by enabling the programmer to declare a type if it makes the code more easy to understand. This is akin of Python's type annotation though it seems a little but more powerful. There is a long documentation about this at [docs.julialang.org/en/v1/manual/types/](https://docs.julialang.org/en/v1/manual/types/). One very definite upside I've learned is that you can only sub type abstract types and extending a concrete type is not possible, as a Spring framework user this warms my mind a lot ;) !
```

Comment:
there is a long documentation -> there is lengthy documentation

Why? Documentation is uncountable noun, it's not one, it's not many. Yet I still think it works with the article, too.